.github/workflows/deploy.yaml

```
@@ -0,0 +1,117 @@
+name: "Deploy"
+
+# Only trigger, when the build workflow succeeded
+on:
+  workflow_call:
+    inputs:
+      environment:
+        description: "Environment to deploy"
+        required: true
+        type: string
+        default: 'staging'
```

actionlint (1.7.8)

input "environment" of workflow_call event has the default value "staging", but it is also required. if an input is marked as required, its default value will never be used

(events)